api/lib/db-interface.js

```diff
@@ -226,6 +243,10 @@ DI.saveTransaction = function(transaction, callback) {
                 + err);
 
     (callback || noop)(err);
+  })
+  .finally(function() {
+    self.lock[txData.client_resource_id] = undefined;
```

Comment:
Realistically the process will restart due to other causes way before this. Even then, any serious deployments will be daemonized/managed